File: js/my_practice/classes/classes.js
class MapObject {
  constructor(latitude, longitude) {
    this.lat = latitude;
    this.long = longitude;
  }

  getMapCoords() {
    return `Latitude: ${this.lat}, Longitude: ${this.long}`;
  }
}

//Subclass(Inheritance)

class GoogleMapMarker extends MapObject {
  constructor(latitude, longitude, label) {
    super(latitude, longitude);
    this.label = label;
  }

  getMapCoords() {
    return `${super.getMapCoords()}, Location: ${this.label}`;
  }
}

// Class Instantiation: To create an object of a class
const marker1 = new GoogleMapMarker(-0.178, 36.980, 'Nairobi')
const marker2 = new GoogleMapMarker(1.235, 37.876, 'Koya')

console.log(marker1.getMapCoords());
console.log(marker2.getMapCoords());

File: python/my_practice/classes/classes.py
from abc import ABC, abstractclassmethod

# Parent Class
class MapObject:
    def __init__(self, latitude, longitude) -> None:
        self.lat = latitude
        self.long = longitude
        
    def get_map_coords(self):
        return f"Latutude: {self.lat}, Longitude: {self.long}"
        
#Subclass (GoogleMapMarker)
class GoogleMapMarker(MapObject):
    def __init__(self, latitude, longitude, label) -> None:
        super().__init__(latitude, longitude)
        self.label = label
        
    def get_map_coords(self):
        return f"{super().get_map_coords()}, Label: {self.label}"
        
# Class Instantiation: Create an object of a class
marker1 = GoogleMapMarker(-0.178, 36.980, 'Nairobi')
marker2 = GoogleMapMarker(1.235, 37.876, 'Koya')

# Output the class call.

print(marker1.get_map_coords())
print(marker2.get_map_coords())